### PR TITLE
NA Enhancements 2 - Electric Bugaloo

### PR DESCRIPTION
**Pull request by cyberkitsune**
+ xgamer.dll downloads correctly now, and no longer gets overwritten with a 0kb file
+ "local config" support
+ small ui edit that doesn't completely break this time

### CODE REVIEW

File: PSO2Launcher/CustomSettingsProvider.cs [hunk 1]
```diff
@@ -198,16 +199,27 @@ private void SaveValuesToFile()
...
-		/// The setting key this is returning must set before the settings are used.
-		/// e.g. <c>Properties.Settings.Default.SettingsKey = @"C:\temp\user.config";</c>
-		/// </summary>
-		private string UserConfigPath => Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.MyDocuments), "SEGA", "dogstar.config");
-
-		/// <summary>
-		/// In memory storage of the settings values
-		/// </summary>
-		private Dictionary<string, SettingStruct> SettingsDictionary { get; set; }
+        /// <summary>
```
Comment (michael-fadely) [inline PSO2Launcher/CustomSettingsProvider.cs]: We're using tabs for indentation. Please convert.

**Comment by cyberkitsune** [comment]
Pending unit tests don't merge yet

File: PSO2Launcher/MainWindow.xaml [hunk 2]
```diff
@@ -511,53 +515,5 @@
...
-                    <Style.Triggers>
-                        <Trigger Property="IsMouseOver" Value="True">
-                            <Setter Property="Fill" Value="{DynamicResource AccentColorBrush}"/>
-                        </Trigger>
-                    </Style.Triggers>
-                </Style>
-            </Rectangle.Style>
-        </Rectangle>
     </Grid>
 </controls:MetroWindow>
```
Comment (michael-fadely) [inline PSO2Launcher/MainWindow.xaml:519]: ~~We had to revert changes to this file to fix the progress bar. Does this button removal also fix the progress bars?~~

I am blind. Disregard.

File: PSO2Launcher/MainWindow.xaml.cs [hunk 3]
```diff
@@ -765,7 +765,7 @@ void cancel(object s, RoutedEventArgs e)
 					}
 
 					PatchListEntry[] lists = launcherListData.Concat(patchListData.Concat(patchListAlways)).ToArray();
-					PatchListEntry[] groups = (from v in lists group v by v.Name into d select d.First()).ToArray();
+                    PatchListEntry[] groups = (from v in lists group v by v.Name into d select d.First()).ToArray();
```
Comment (michael-fadely) [inline PSO2Launcher/MainWindow.xaml.cs]: Tabs for indentation please.

File: PSO2Launcher/MainWindow.xaml.cs [hunk 4]
```diff
@@ -846,8 +846,8 @@ void pat(Task t)
 							if (data.Source == PatchListSource.None)
 							{
 								var fakeSource = PatchListSource.Master;
-
-								if (patchListData.Contains(data) || launcherListData.Contains(data))
+                                // HACK: Type check is a sloppy hack for NA -- all patches come from Patch on NA, not master. (As far as we can tell?)
```
Comment (michael-fadely) [inline PSO2Launcher/MainWindow.xaml.cs]: Tabs for indentation please.

**Review by michael-fadely — CHANGES_REQUESTED** [review]

**Review by michael-fadely — APPROVED** [review]

**Pull request merged**